pkg/buildkit/worker/pool.go

```diff
@@ -375,8 +376,7 @@ func (p *workerPool) updateWorkers(ctx context.Context) error {
 			if req := p.requests.Dequeue(); req != nil {
 				log.Info("Found pending pod request, processing")
 
-				pod := pod
-				if p.processPodRequest(ctx, req, &pod) {
+				if p.processPodRequest(ctx, req, pod) {
```

Comment:
Just passing the pod value instead of a pointer to avoid implicit memory aliasing and variable re-declaration.